conda_smithy/templates/circle.yml.tmpl

```diff
@@ -24,14 +24,31 @@ dependencies:
   override:
     - docker pull condaforge/linux-anvil
 
+{%- if matrix %}
+{%- if 'ccache-toolchain' in ' '.join(package['meta']['requirements']['build']) %}
```

Comment:
Yes, if we moved the code to `toolchain` then this `if` check would check for `toolchain`.

I had it this way, so that we can disable caching in different platforms simply by using a selector on `ccache-toolchain`.